### ISSUE TITLE
add  `OGRGeometry MakeValid` function

### ISSUE THREAD

**Issue by wandergis** [mmomtchev/node-gdal-async#8]
![image](https://user-images.githubusercontent.com/5799374/124454765-68b13980-ddbb-11eb-9158-a5bbea0ef4b6.png)

**Comment by mmomtchev** [comment]
@wandergis, can you attach a file with a geometry that will be affected by this function?

**Comment by wandergis** [comment]
@mmomtchev  Here is the geojson data
```js
{
  "type": "MultiPolygon",
  "coordinates": [
    [
      [
        [111.046746, 33.202932],
        [111.032431, 33.209355],
        [111.030368, 33.21456],
        [111.024992, 33.219494],
        [111.020408, 33.220341],
        [111.016728, 33.223975],
        [111.014207, 33.229993],
        [111.011135, 33.234112],
        [111.005771, 33.235469],
        [110.868439, 33],
        [111.08881, 33.182041],
        [111.080879, 33.181917],
        [111.073245, 33.18065],
        [111.061887, 33.174028],
        [111.040053, 33.156915],
        [111.03721, 33.160988],
        [111.045772, 33.170079],
        [111.033886, 33.178051],
        [111.029691, 33.174917],
        [111.026494, 33.175139],
        [111.024923, 33.173568],
        [111.020557, 33.17235],
        [111.016557, 33.172613],
        [111.015812, 33.174036],
        [111.019112, 33.176932],
        [111.024029, 33.17991],
        [111.025279, 33.182049],
        [111.028442, 33.184369],
        [111.030459, 33.187725],
        [111.033199, 33.188802],
        [111.036018, 33.187955],
        [111.046746, 33.202932]
      ]
    ],
    [
      [
        [111.046746, 33.202932],
        [111.036018, 33.187955],
        [111.037806, 33.184805],
        [111.035308, 33.182148],
        [111.035697, 33.180667],
        [111.033886, 33.178051],
        [111.045772, 33.170079],
        [111.057829, 33.183587],
        [111.056534, 33.192734],
        [111.051434, 33.200671],
        [111.046746, 33.202932]
      ]
    ]
  ]
}
```
### before
![image](https://user-images.githubusercontent.com/5799374/124462989-d4e46b00-ddc4-11eb-96c0-11927af15648.png)


### after
![image](https://user-images.githubusercontent.com/5799374/124462859-a8305380-ddc4-11eb-9856-231e0ca72004.png)

**Comment by mmomtchev** [comment]
Landed in 3.3.2
